### DISCUSSION TITLE
Cross Platform Desktop MessageBoxes.

### DISCUSSION THREAD

**Discussion by AraHaan** [dotnet/maui#1120] [Ideas]
I currently use Windows Forms and in most of my programs I use the windows MessageBox APIs.

These help in cases where I want to prompt the user to choose an option I want them to select and filter out if they pressed:
- Ok
- Cancel
- Any other button I could set to display on said MessageBox.

For Windows Forms that is easy, however if I wanted the application to work for Windows, Mac, and Linux for a single codebase I would like to be able to have something similar in MAUI.

While it can be argued that for Windows my applications should use the newer style of MessageBoxes that allows to customize the whole thing, it's also not as simple as the older one.

MessageBoxes should be supported on Mac and Linux as well easily it just depends on how to actually make them on a Mac and ensure they are created just like how they are created on Windows.

**Comment by Happypig375** [comment]
This can be shown as alerts when on iOS and Android.

**Comment by AmSmart** [comment]
Each Platform handles these scenarios differently, however, some cross-platform ``Views`` like the ``Snackbar``, ``Toast`` and ``Popup`` which are all in the Xamarin Community Toolkit can handle most, if not all of your scenarios.
